src/piqtree2/iqtree/_tree.py

```diff
@@ -134,12 +134,15 @@ def _process_tree_yaml(
     newick = tree_yaml["PhyloTree"]["newick"]
 
     tree = cogent3.make_tree(newick)
-
+    t2t_matrix = tree.tip_to_tip_distances()[0]
     candidates = tree_yaml["CandidateSet"]
     likelihood = None
     for candidate in candidates.values():
-        if newick in candidate:
-            likelihood = float(candidate.split(" ")[0])
+        candidate_likelihood, candidate_newick = candidate.split(" ")
+        candidate_tree = cogent3.make_tree(candidate_newick)
+        candidate_t2t_matrix = candidate_tree.tip_to_tip_distances()[0]
+        if np.allclose(t2t_matrix, candidate_t2t_matrix):
+            likelihood = float(candidate_likelihood)
             break
     if likelihood is None:
         msg = "IQ-TREE output malformated, likelihood not found"
```

Comment:
This became uncovered because my changed newick string in the test only changed after about the 10th decimal place, covered again now but there's a problem.